xds/src/main/java/io/grpc/xds/client/XdsClientImpl.java

```diff
@@ -261,35 +248,111 @@ public void run() {
         }
         ResourceSubscriber<T> subscriber =
             (ResourceSubscriber<T>) resourceSubscribers.get(type).get(resourceName);
+
         if (subscriber == null) {
           logger.log(XdsLogLevel.INFO, "Subscribe {0} resource {1}", type, resourceName);
           subscriber = new ResourceSubscriber<>(type, resourceName);
           resourceSubscribers.get(type).put(resourceName, subscriber);
-          if (subscriber.controlPlaneClient != null) {
-            subscriber.controlPlaneClient.adjustResourceSubscription(type);
+
+          CpcWithFallbackState cpcToUse = manageControlPlaneClient(subscriber);
+          if (cpcToUse.cpc != null) {
```

Comment:
When the retry timer goes off and the CPC tries to reconnect, if it fails, then handleStreamClosed will call onError (the same as previously done).  It seems like we shouldn't continue sending errors for every backoff, but that should probably be addressed in a separate PR.

Reply:
Yes, I had considered that, but retries can take 2 minutes. It is known to be failing at this very moment, so we shouldn't delay an error.

I think it is proper to update the error each backoff; the error can change over time.

Reply:
If we don't care about an extra call to `onError`, then it trivially becomes adding a call to `subscriber.onError` in the `catch (IOException)` of `manageControlPlaneClient`

Reply:
Why `manageControlPlaneClient()`, which I've already said is broken to pass a subscriber, when we could just call `onError()` directly here? Nothing actually knows the error more than "no working cpcs." I was most interested in having an actionable error message, but having a call to onError() here would be fine except...

... we'd have to be careful about it to avoid https://github.com/grpc/grpc-java/issues/11672 . Let's not worry about it in this PR, because it is pre-existing, broken in multiple ways, and fallback is enough trouble by itself.